server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -179,6 +179,11 @@ public static ThreadPoolType fromType(String type) {
 
     public static final double searchAutoscalingEWMA = 0.1;
 
+    // This value is chosen such that a sudden increase in the task durations would need to persist roughly for 60 samples
+    // for the EWMA value to be mostly representative of the increased task durations. Mostly representative means that the
+    // EWMA value is at least within 90% of the new increased task duration.
```

Comment:
When I read this, I understood what it means but failed to see the immediate impact. So I find it useful to add something related to Henning's [example](https://github.com/elastic/elasticsearch/pull/112206#issuecomment-2312004755), i.e. a single large value bumps the average to at least `indexAutoscalingEWMA *  value`, e.g. for 20 seconds, this is 2.0 seconds with the existing alpha of 0.1. I think the explanation on immediate impact (emphasing on how we want to smooth out a sudden spike) compliments the longer term impact (90% after 60 samples) which  emphasizes on how quickly actual increase in execution time can be reflected.